# Developed in house to disable MD4
# Internal, will not be submitted upstream.

--- php-5.6.8/ext/openssl/openssl.c.~1~	Wed Apr 15 11:05:57 2015
+++ php-5.6.8/ext/openssl/openssl.c	Wed Apr 13 08:26:50 2016
@@ -64,7 +64,9 @@
 
 #define OPENSSL_ALGO_SHA1 	1
 #define OPENSSL_ALGO_MD5	2
+#ifdef	HAVE_OPENSSL_MD4_H
 #define OPENSSL_ALGO_MD4	3
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 #define OPENSSL_ALGO_MD2	4
 #endif
@@ -1027,9 +1029,11 @@
 		case OPENSSL_ALGO_MD5:
 			mdtype = (EVP_MD *) EVP_md5();
 			break;
+#ifdef	HAVE_OPENSSL_MD4_H
 		case OPENSSL_ALGO_MD4:
 			mdtype = (EVP_MD *) EVP_md4();
 			break;
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 		case OPENSSL_ALGO_MD2:
 			mdtype = (EVP_MD *) EVP_md2();
@@ -1151,7 +1155,9 @@
 	/* signature algorithm constants */
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_SHA1", OPENSSL_ALGO_SHA1, CONST_CS|CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD5", OPENSSL_ALGO_MD5, CONST_CS|CONST_PERSISTENT);
+#ifdef HAVE_OPENSSL_MD4_H
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD4", OPENSSL_ALGO_MD4, CONST_CS|CONST_PERSISTENT);
+#endif
 #ifdef HAVE_OPENSSL_MD2_H
 	REGISTER_LONG_CONSTANT("OPENSSL_ALGO_MD2", OPENSSL_ALGO_MD2, CONST_CS|CONST_PERSISTENT);
 #endif
